api/src/main/java/me/lokka30/treasury/api/economy/account/Account.java

```diff
@@ -152,4 +263,91 @@ public void fail(@NotNull EconomyException exception) {
+     *
+     * <p>If the specified {@code transactionCount} is higher than the known transactions, then this will return all the
+     * transactions.
+     * <p>If the specified {@code transactionCount} is {@code -1}, then this will return all the transactions. <i>If you
+     * really need all the transactions, please use this with care, because an economy provider might do a query to a database
+     * .</i>
+     *
+     * @param transactionCount the count of the transactions wanted
+     * @param subscription     the {@link EconomySubscriber} accepting the transaction history
+     * @author MrIvanPlays
+     * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
+     */
+    void retrieveTransactionHistory(
+            int transactionCount, @NotNull EconomySubscriber<Collection<EconomyTransaction>> subscription
+    );
+
+    /**
+     * Request the {@link EconomyTransaction} history, starting from the {@link Temporal} {@code from} specified, of this
+     * {@code Account}.
+     *
+     * @param from         the timestamp to get the transactions from
+     * @param subscription the {@link EconomySubscriber} accepting the transaction history
+     * @author MrIvanPlays
+     * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
+     */
+    default void retrieveTransactionHistory(
+            @NotNull Temporal from, @NotNull EconomySubscriber<Collection<EconomyTransaction>> subscription
+    ) {
+        retrieveTransactionHistory(from, Instant.now(), subscription);
+    }
+
+    /**
+     * Request the {@link EconomyTransaction} history, starting from the {@link Temporal} {@code from} specified and
+     * ending at {@link Temporal} {@code to} specified, of this {@code Account}.
+     *
+     * @param from         the timestamp to get the transactions from
+     * @param to           the timestamp to get the transactions to
+     * @param subscription the {@link EconomySubscriber} accepting the transaction history
+     * @author MrIvanPlays
+     * @since {@link me.lokka30.treasury.api.economy.misc.EconomyAPIVersion#v1_0 v1.0}
+     */
+    default void retrieveTransactionHistory(
+            @NotNull Temporal from, @NotNull Temporal to, @NotNull EconomySubscriber<Collection<EconomyTransaction>> subscription
+    ) {
+        Objects.requireNonNull(from, "from");
+        Objects.requireNonNull(to, "to");
+        Objects.requireNonNull(subscription, "subscription");
+
+        Instant fromInstant;
+        if (from instanceof Instant) {
+            fromInstant = (Instant) from;
+        } else {
+            fromInstant = Instant.from(from);
+        }
+        Instant toInstant;
+        if (to instanceof Instant) {
+            toInstant = (Instant) to;
+        } else {
+            toInstant = Instant.from(to);
+        }
+        retrieveTransactionHistory(-1, new EconomySubscriber<Collection<EconomyTransaction>>() {
```

Comment:
Retrieving the entirety of transaction history and then filtering it is horrendously inefficient, it defeats the primary purpose of loading only a segment of the history. The implementation should handle this method in its entirety so that unnecessary transaction entries are not loaded at all.